control/views.py

```diff
@@ -144,12 +144,15 @@ def currency_converter(request):
             unit_to = form.cleaned_data['unit_to'].upper()
             unit_value = form.cleaned_data['unit_value']
 
-            url = 'https://api.exchangeratesapi.io/latest?base='+unit
-            currency_values = requests.get(url).json()
-            currency_value = currency_values['rates'][unit_to]
-
-            calculations = unit_value * currency_value
-            result = '{} {}'.format(calculations, unit_to)
+            try:
+                url = 'https://api.exchangeratesapi.io/latest?base='+unit
+                currency_values = requests.get(url).json()
+                currency_value = currency_values['rates'][unit_to]
```

Comment:
When `try...except..` an error, you have to be as much specific as you can. In this case, this error is only affecting the line: `currency_value = currency_values['rates'][unit_to]`  so...

```suggestion
                url = 'https://api.exchangeratesapi.io/latest?base='+unit
                currency_values = requests.get(url).json()
            try:
                currency_value = currency_values['rates'][unit_to]
            except KeyError:
                currency_value = 1
            else:
                calculations = unit_value * currency_value
                result = '{} {}'.format(calculations, unit_to)
```

Reply:
Ok, I get it it's more clean this way, what I don't really understand is the else: on line 156, that makes the code ignore the formula and it will always return 0, even if you try to convert 1 euro to eur, it should return 1 not 0. I've omited the else and with the right indent works fine I'll push it so you can see it.

Reply:
Yes, it should be an `else` clause, it should be `finally` instead. Good shot

Reply:
it should? or it shouldn't? haha you're confusing me

Reply:
lol shouldn't